### PR TITLE
Update tokens based on new theming convention

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #227, #229, #236

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Update token names to use new `--ds` prefix
* Update dependencies
* Update to support Node 20
* Update branch references
* Incorporate the use of the updated `auro-alert`
* Update hover colors to use linear gradient
* Update JSDocs
* Remove focus-visible support

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ branches:
         # Required. Require branches to be up to date before merging.
         strict: true
         # Required. The list of status checks to require in order to merge into this branch
-        contexts: ["test (18.x)", "license/cla"]
+        contexts: ["test (20.x)", "license/cla"]
```
Comment (blackfalcon) [inline .github/settings.yml:59]:
```suggestion
        contexts: ["test (18.x)", "test (20.x)", "license/cla"]
```

File: .github/workflows/publishDemo.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Deploy Demo
 
 on:
   pull_request:
-    branches: [ main ]
+    branches: [ master ]
```
Comment (blackfalcon) [inline .github/workflows/publishDemo.yml:5]: Why was this changed?

File: demo/api.html [hunk 3]
```diff
@@ -9,16 +9,14 @@
       type="text/css"
       href="https://cdn.jsdelivr.net/npm/prismjs@latest/themes/prism.css"
     />
-    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
```
Comment (blackfalcon) [inline demo/api.html:12]: Please update this section to mirror this code example. 

https://github.com/AlaskaAirlines/auro-avatar/blob/main/demo/index.html#L12-L17

File: demo/api.html [hunk 4]
```diff
@@ -34,6 +32,9 @@
     </script>
     <script src="https://cdn.jsdelivr.net/npm/@alaskaairux/auro-accordion@latest/dist/auro-accordion__bundled.js" type="module"></script>
```
Comment (blackfalcon) [inline demo/api.html:33]: Please update the repo to support the new accordion

File: demo/index.html [hunk 5]
```diff
@@ -9,16 +9,14 @@
       type="text/css"
       href="https://cdn.jsdelivr.net/npm/prismjs@latest/themes/prism.css"
     />
-    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
-    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@alaskaairux/webcorestylesheets@latest/dist/demoWrapper.css" />
-    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@alaskaairux/webcorestylesheets@latest/dist/elementDemoStyles.css" />
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
+    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/webcorestylesheets@latest/dist/demoWrapper.css" />
+    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/webcorestylesheets@latest/dist/elementDemoStyles.css" />
```
Comment (blackfalcon) [inline demo/index.html:14]: Please update this section to mirror this code 

https://github.com/AlaskaAirlines/auro-avatar/blob/main/demo/index.html#L12-L17

File: demo/index.html [hunk 6]
```diff
@@ -30,6 +28,9 @@
     <script type="module" src="../index.js"></script>
     <script src="https://cdn.jsdelivr.net/npm/@alaskaairux/auro-accordion@latest/dist/auro-accordion__bundled.js" type="module"></script>
```
Comment (blackfalcon) [inline demo/index.html:29]: Update to support the new accordion

File: package.json [hunk 7]
```diff
@@ -17,57 +17,54 @@
   "main": "index.js",
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=18"
+    "node": "^18 || ^20"
   },
   "dependencies": {
     "@alaskaairux/auro-loader": "^1.1.1",
+    "@aurodesignsystem/design-tokens": "^4.1.1",
+    "@aurodesignsystem/webcorestylesheets": "^5.0.4",
```
Comment (blackfalcon) [inline package.json:25]: These should be peer dependencies. Should be like 

```
  "peerDependencies": {
    "@aurodesignsystem/design-tokens": "",
    "@aurodesignsystem/webcorestylesheets": ""
  },
  "devDependencies": {
    "@aurodesignsystem/design-tokens": "",
    "@aurodesignsystem/webcorestylesheets": "",
    "@aurodesignsystem/eslint-config": "",
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: .github/workflows/publishDemo.yml (hunk 2 above)
Comment (jordanjones243) [reply .github/workflows/publishDemo.yml:5]: This was changed because button uses `master` not `main`

File: package.json (hunk 7 above)
Comment (jordanjones243) [reply package.json:25]: What is `@aurodesignsystem/eslint-config`? Was that a package I was supposed to add?

**Review by blackfalcon — APPROVED** [review]

File: src/style.scss [hunk 8]
```diff
@@ -40,47 +40,47 @@ $auro-inset-directions:'--squish';
...
 $auro-rgb-color-base-white-03: rgba(255, 255, 255, .03);
 
-@import "./../node_modules/@aurodesignsystem/webcorestylesheets/dist/utilityMixins/insetUtility";
+@import "./../node_modules/@aurodesignsystem/webcorestylesheets/src/utilityMixins/insetUtility";
 
 // focus-visible mixin
 @mixin focus-visible {
   outline: 3px solid transparent;
-  box-shadow: inset 0 0 0 1px var(--auro-color-border-focus-on-dark), inset 0 0 0 3px var(--auro-color-background-lightest);
+  box-shadow: inset 0 0 0 1px var(--ds-color-border-focus-inverse), inset 0 0 0 3px var(--ds-color-background-lightest);
```
Comment (blackfalcon) [inline src/style.scss:48]: All tokens should have the Sass variable fallback.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: demo/api.md [hunk 9]
```diff
@@ -46,12 +46,12 @@
...
-| [button](#button)      | Apply CSS to HTML5 button. |
-| `button-icon` | Apply CSS to icon slot.    |
-| `button-text` | Apply CSS to text slot.    |
-| [loader](#loader)      | Apply CSS to auro-loader.  |
+| Part     | Description                |
+|----------|----------------------------|
+| [button](#button) | Apply CSS to HTML5 button. |
+| [icon](#icon)   | Apply CSS to icon slot.    |
+| [loader](#loader) | Apply CSS to auro-loader.  |
+| [text](#text)   | Apply CSS to text slot.    |
```
Comment (jason-capsule42) [inline demo/api.md:54]: Changing the part names would make this a breaking change. 

This change isn't justified in the linked tickets or the PR description.

File: src/style.scss [hunk 10]
```diff
@@ -40,47 +40,47 @@ $auro-inset-directions:'--squish';
...
   // auro-button--tertiary
   &[variant='tertiary'],
   &--tertiary {
-    background-color: var(--auro-color-background-lighter);
-    border-color: var(--auro-color-ui-default-on-light);
-    box-shadow: inset 0 0 0 3px var(--auro-color-ui-default-on-light);
+    background-color: var(--ds-color-background-lighter, $ds-color-background-lighter);
+    border-color: var(--ds-color-ui-default-default, $ds-color-ui-default-default);
+    box-shadow: inset 0 0 0 3px var(--ds-color-ui-default-default, $ds-color-ui-default-default);
   }
```
Comment (jason-capsule42) [inline src/style.scss:63]: All focus-visible references can be removed. This is an old utility we don't support anymore.

File: src/style.scss [hunk 11]
```diff
@@ -40,47 +40,47 @@ $auro-inset-directions:'--squish';
...
 
     // auro-buttonOndark--secondary
     &[variant='secondary'],
     &--secondary {
-      background-color: var(--auro-color-base-black-opacity-15);
-      box-shadow: inset 0 0 0 3px var(--auro-color-ui-active-on-dark);
-      border: 1px solid var(--auro-color-ui-active-on-dark);
+      background-color: var(--ds-color-base-black-opacity-15, $ds-color-base-black-opacity-15);
+      box-shadow: inset 0 0 0 3px var(--ds-color-brand-breeze-200, $ds-color-brand-breeze-200);
+      border: 1px solid var(--ds-color-brand-breeze-200, $ds-color-brand-breeze-200);
```
Comment (jason-capsule42) [inline src/style.scss:75]: I'm not seeing in the tickets linked to change to the brand breeze color.

File: src/style.scss [hunk 12]
```diff
@@ -40,47 +40,47 @@ $auro-inset-directions:'--squish';
...
 
     // auro-buttonOndark--tertiary
     &[variant='tertiary'],
     &--tertiary {
-      background-color: var(--auro-color-base-black-opacity-15);
-      border-color: var(--auro-color-ui-active-on-dark);
-      box-shadow: inset 0 0 0 3px var(--auro-color-ui-active-on-dark);
+      background-color: var(--ds-color-base-black-opacity-15, $ds-color-base-black-opacity-15);
+      border-color: var(--ds-color-brand-breeze-200, $ds-color-brand-breeze-200);
+      box-shadow: inset 0 0 0 3px var(--ds-color-brand-breeze-200, $ds-color-brand-breeze-200);
```
Comment (jason-capsule42) [inline src/style.scss:83]: I'm not seeing in the tickets linked to change to the brand breeze color.

File: src/style.scss [hunk 13]
```diff
@@ -276,65 +276,65 @@ slot {
...
+        background-image: linear-gradient(var(--ds-color-ui-hover-inverse, $ds-color-ui-hover-inverse), var(--ds-color-ui-hover-inverse, $ds-color-ui-hover-inverse));
+        border: 1px solid var(--ds-color-ui-hover-inverse, $ds-color-ui-hover-inverse);
       }
     }
 
     &:active {
-      background-color: var(--auro-color-ui-active-on-dark);
-      border: 1px solid var(--auro-color-ui-active-on-dark);
+      background-color: var(--ds-color-brand-breeze-200, $ds-color-brand-breeze-200);
+      border: 1px solid var(--ds-color-brand-breeze-200, $ds-color-brand-breeze-200);
```
Comment (jason-capsule42) [inline src/style.scss:294]: I'm not seeing in the tickets linked to change to the brand breeze color.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: demo/api.md (hunk 9 above)
Comment (jason-capsule42) [reply demo/api.md:54]: Turns out this was just a documentation issue. Previous release did not rebuild docs.

File: src/style.scss (hunk 13 above)
Comment (jason-capsule42) [reply src/style.scss:247]: @leeejune after some conversation with @jordanjones243 he switched to breeze because the original token here was disabled but the brand breeze color had the same hex code. 

Can you update us here what the correct token for the figma design should be under the new DS tokens?

Comment (leeejune) [reply src/style.scss:247]: I've added a few missing design tokens in this issue. This should solve this conflict. https://github.com/AlaskaAirlines/AuroDesignTokens/issues/137

File: src/style.scss [hunk 14]
```diff
@@ -150,15 +103,15 @@ slot {
   @media (hover: hover) {
     &:active,
     &:hover {
-      background-color: var(--auro-color-ui-hover-on-light);
```
Comment (jason-capsule42) [inline src/style.scss:153]: I don't remember why we made this change as part of this PR? The PR description says it was done but not why. Please clarify why the change was made in the description.

File: src/style.scss [hunk 15]
```diff
@@ -189,50 +142,50 @@ slot {
...
     }
 
     &:disabled {
       cursor: not-allowed;
       transform: unset;
 
-      color: rgba($auro-color-text-link-on-light, $disabledOPacity);
-      background-color: rgba($auro-color-background-lightest, $disabledOPacity);
-      border: 1px solid rgba($auro-color-ui-default-on-light, $disabledOPacity);
+      color: rgba($ds-color-text-link-default, $disabledOPacity);
```
Comment (jason-capsule42) [inline src/style.scss:180]: These colors that don't use tokens are not able top be themed. Wasn't part of this work intended to get new tokens so at least all colors are theme-able?

File: src/style.scss (hunk 14 above)
Comment (jordanjones243) [reply src/style.scss:153]: We had to change from `background-color` to `background-image` for the hover styles in order to allow for `linear-gradient`

File: src/style.scss (hunk 15 above)
Comment (jordanjones243) [reply src/style.scss:180]: From my understanding, these variables are being consumed from our sass variables, so they should still be theme-able if an implementer creates their own sass files with their own values.

File: src/style.scss (hunk 14 above)
Comment (jordanjones243) [reply src/style.scss:153]: These were changed from `background-color` to `background-image` in order to work together with the `linear-gradient` we wanted to put on all hover styles

**Comment by jason-capsule42** [comment]
This PR is approved pending @jordanjones243 opening a ticket for the non-themeable opacity colors called out in earlier comments.

**Review by leeejune — APPROVED** [review]

**Comment by jordanjones243** [comment]
> This PR is approved pending @jordanjones243 opening a ticket for the non-themeable opacity colors called out in earlier comments.

The issue for this problem has been created:
- https://github.com/AlaskaAirlines/auro-button/issues/237
